### DISCUSSION TITLE
How to fix plugin.xml problem of Teamcenter

### DISCUSSION THREAD

**Discussion by mq1277515053** [JaneWardSandy/eclipse-pde-partial-idea#97] [Q&A] [answered]
Idea can not recognize XML file from eclipse teamcenter, which DTD file should I import?
![image](https://user-images.githubusercontent.com/98294069/207542057-e34cc460-23aa-4e68-8d25-4f21ce35d04b.png)
![image](https://user-images.githubusercontent.com/98294069/207542132-89bebe99-764e-4755-9ac6-a3bfe852ed1e.png)

**Comment by JaneWardSandy** [comment, marked as answer]
You should extract schema directory on `org.eclipse.core.expressions.source_version.jar` and `org.eclipse.ui.source_version.jar` from eclipse SDK, then put them into the corresponding jar in `Teamcenter/plugins`, then resolve the workspace and it should be work well.

see also #44 #53 
<img width="1105" alt="image" src="https://user-images.githubusercontent.com/15345355/207543603-c69c2dcc-464c-4016-acc2-9ee83610f953.png">

**Reply by mq1277515053** [reply]
Thanks，I will try it！

**Reply by mq1277515053** [reply]
It is stranger，the two JARs do not contain the schema directory.

**Reply by JaneWardSandy** [reply]
Teamcenter was not include the Eclipse SDK source jar, so you need to get it by yourself.
Or, using the attachment which was I'm using with.
[schema.zip](https://github.com/JaneWardSandy/eclipse-pde-partial-idea/files/10242366/schema.zip)

**Reply by mq1277515053** [reply]
![image](https://user-images.githubusercontent.com/98294069/208013981-586c07bf-c343-4a33-9484-4e62f66be26c.png)
![image](https://user-images.githubusercontent.com/98294069/208014054-318bf677-85cf-4146-9289-a36c9df4fc57.png)

**Comment by mq1277515053** [comment]
It is working, Thanks very much!  Why I can not find the schema directory for eclipse SDK?

**Reply by JaneWardSandy** [reply]
Generally speaking, the Eclipse IDE we downloaded does not contain the source code, and we need to download the code package separately

like this https://github.com/JaneWardSandy/eclipse-pde-partial-idea/issues/44#issuecomment-891035159

**Reply by JaneWardSandy** [reply]
`org.eclipse.ui_version.jar` was binary package.
`schema` directory was included in `org.eclipse.ui.source_version.jar`, `*.source_version.jar` was included in `Eclipse SDK` and `Eclipse Platform SDK`.
And we download the `Eclipse` in eclipse home website, it was `Eclipse IDE`, not `Eclipse SDK`

**Reply by mq1277515053** [reply]
Fine, thanks! I got it.
